pulp_smash/tests/docker/cli/test_copy.py

```diff
@@ -0,0 +1,244 @@
+            raise unittest2.SkipTest('These tests require Pulp 2.8 or above.')
+        # Create and sync a repository with a v1 feed to bring some Images in.
+        docker_utils.create_repo(
+            cls.cfg, cls.repo_id, upstream_name='library/busybox',
+            sync_v1=True, sync_v2=False)
+        docker_utils.sync_repo(cls.cfg, cls.repo_id)
+
+        # Now create a feedless repository and copy the images from the first
+        # repository over.
+        cls.copy_target = utils.uuid4()
+        docker_utils.create_repo(cls.cfg, cls.copy_target)
+        cls.completed_proc = docker_utils.copy(cls.cfg, 'image', cls.repo_id,
+                                               cls.copy_target)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete the copy_target repo."""
+        super(CopyAllImagesTestCase, cls).tearDownClass()
+        docker_utils.delete_repo(cls.cfg, cls.copy_target)
+
+    def test_positive_copy_output(self):
+        """Assert that the list of image_ids in the copy output are correct."""
+        # We need to limit the fields to image_id as a hack to work around
+        # https://pulp.plan.io/issues/1696. Doing this has a side effect of
+        # showing the image_ids without splitting them with a line break
+        # (sigh).
+        units_in_src = docker_utils.search(self.cfg, 'image', self.repo_id,
+                                           fields=['image_id']).stdout
+        # Due to https://pulp.plan.io/issues/1693 it is not possible to get
+        # pulp-admin to limit the output to the image_ids. Hello regex my old
+        # friend, I've come to talk with you again…
+        image_ids_in_src = set(re.findall(r'(?:Image Id:)\s*(.*)',
+                                          units_in_src))
+
+        image_ids_printed = set(re.findall(r'(?: {2})(\w*)',
+                                           self.completed_proc.stdout))
+
+        self.assertEqual(image_ids_printed, image_ids_in_src)
+
+    def test_positive_units_copied(self):
+        """Assert that all units were copied."""
+        units_in_src = docker_utils.search(self.cfg, 'image',
+                                           self.repo_id).stdout
+        units_in_dest = docker_utils.search(self.cfg, 'image',
+                                            self.copy_target).stdout
+
+        # Due to https://pulp.plan.io/issues/1693 it is not possible to get
+        # pulp-admin to limit the output to the unit_ids. Hello regex my old
+        # friend, I've come to talk with you again…
+        units_in_src = set(re.findall(r'(?:Unit Id:)\s*(.*)', units_in_src))
+        units_in_dest = set(re.findall(r'(?:Unit Id:)\s*(.*)', units_in_dest))
+
+        self.assertEqual(units_in_src, units_in_dest)
+
+    def test_return_code(self):
+        """Assert that the copy command has a return code of 0."""
+        self.assertEqual(self.completed_proc.returncode, 0)
+
+    def test_task_succeeded(self):
```

Comment:
This same assertion appears in all three test cases. How about adding this to `SuccessMixin`?